Enhance removing promoted jobs experience

renatho opened the pull request:
Fixes #2546

### Changes proposed in this Pull Request

* Cancel promoted jobs deletion.
* Add a notice when there are promoted jobs on trash.
* Notice that tied to this, we also have [this PR](https://github.com/Automattic/WP-Job-Manager/pull/2544), making sure that users can deactivate jobs from trash, and we [highlight the promoted badge](https://github.com/Automattic/WP-Job-Manager/pull/2544#issuecomment-1652334761).

### Testing instructions

* Create 2 jobs, and promote 1 of them.
* Send both jobs to trash.
* Check that a notice is displayed that you have promoted jobs on the trash.
* Click on the button to navigate to the trash.
* Try to delete the jobs permanently. Make sure you can't delete the promoted job.
* Restore the promoted job, and make sure the notice is not displayed anymore.
* Try to create normal posts and delete them permanently, and make sure it continues working properly.

<!--
Helpful tips for screenshots:
https://en.support.wordpress.com/make-a-screenshot/
-->
### Screenshot / Video

<img width="1460" alt="Screenshot 2023-07-26 at 17 09 02" src="https://github.com/Automattic/WP-Job-Manager/assets/876340/69afa847-3f2b-41cb-8ac7-d00c7cd8aedc">

renatho committed:
Avoid deleting promoted jobs

renatho committed:
Update access private to internal

renatho committed:
Add method to count promoted jobs with specific args

renatho committed:
Add admin notice when there are promoted jobs on trash

renatho commented on includes/admin/class-wp-job-manager-promoted-jobs-admin.php:
```diff
@@ -390,6 +391,52 @@ public function maybe_add_promoted_jobs_notice( $notices ) {
+
+		$notices['promoted-job-on-trash'] = [
+			'type'        => 'user',
+			'dismissible' => false,
```
I made it not dismissable, so we make sure it will show up every time the user has a promoted job on trash.

aaronfc commented on includes/admin/class-wp-job-manager-promoted-jobs-admin.php:
```diff
@@ -390,6 +391,52 @@ public function maybe_add_promoted_jobs_notice( $notices ) {
+			'type'        => 'user',
+			'dismissible' => false,
+			'level'       => 'info',
+			'heading'     => __( 'You have promoted jobs on trash.', 'wp-job-manager' ),
```
Minor thing but is it "on trash" or "*in* (the?) trash" ?

renatho replied:
https://github.com/Automattic/WP-Job-Manager/pull/2548/commits/ce74c0e1ec1fd72becab4658414111a1801fe2ca

aaronfc commented on includes/admin/class-wp-job-manager-promoted-jobs-admin.php:
```diff
@@ -390,6 +391,52 @@ public function maybe_add_promoted_jobs_notice( $notices ) {
+			'dismissible' => false,
+			'level'       => 'info',
+			'heading'     => __( 'You have promoted jobs on trash.', 'wp-job-manager' ),
+			'message'     => __( 'Deactivate trashed jobs in order to stop the promotion or publish it again, so applicants can see the job on your site.', 'wp-job-manager' ),
```
I am not sure about the "Deactivate trashed jobs" part... This is my re-wording intent:
"Trashed jobs are not be available to your applicants. Deactivate the promotion or publish the job again to fix this."

renatho replied:
https://github.com/Automattic/WP-Job-Manager/pull/2548/commits/ce74c0e1ec1fd72becab4658414111a1801fe2ca

aaronfc reviewed (commented):
Works great! I left a couple of minor comments about wording. Feel free to change them or not :)

---

Given that the error message is quite ugly when trying to "Delete permanently" a promoted job, what if we remove the action?

I got it working with this (but I am not 100% sure if it's the best approach, might need a bit more of testing):
```
add_filter( 'post_row_actions', function ( $actions, $post ) {
	if (WP_Job_Manager_Promoted_Jobs::is_promoted($post->ID)) {
		unset($actions['delete']);
	}

    return $actions;
}, 10, 2 );
```

---

Also, I am wondering... What happens with the "automatic" process for emptying the trash? I gave a [quick look at the code](https://github.com/WordPress/wordpress-develop/blob/6.2/src/wp-includes/functions.php#L6587-L6627) and it looks like they will call the normal `wp_delete_post` method. But they don't check for the response so I think we are safe?

renatho committed:
Remove unecessary check

Already checked as first thing on `is_promoted`

renatho committed:
Updade delete link from post row to not be allowed

renatho committed:
Use WP Admin tooltip for the title

renatho committed:
Improve copy

renatho commented:
> I left a couple of minor comments about wording

Thank you for the suggestions! I think it looks better now! https://github.com/Automattic/WP-Job-Manager/pull/2548/commits/ce74c0e1ec1fd72becab4658414111a1801fe2ca

> Given that the error message is quite ugly when trying to "Delete permanently" a promoted job, what if we remove the action?

Good idea! I just did it a little different, and added a tooltip explaining the reason. I also used the data-tooltip you mentioned in the other PR. It happens in the job listing and also in the trash. WDYT? https://github.com/Automattic/WP-Job-Manager/pull/2548/commits/3d8896da4b0c89c18d17b974183537fe44ef32bf

<img width="296" alt="Screenshot 2023-07-28 at 11 21 12" src="https://github.com/Automattic/WP-Job-Manager/assets/876340/70771517-5e02-4323-80d0-113d46cc0491">

> Also, I am wondering... What happens with the "automatic" process for emptying the trash? I gave a [quick look at the code](https://github.com/WordPress/wordpress-develop/blob/6.2/src/wp-includes/functions.php#L6587-L6627) and it looks like they will call the normal `wp_delete_post` method. But they don't check for the response so I think we are safe?

Just to make sure, I updated the `_wp_trash_meta_time` in the database of one promoted job and one not promoted. So I ran the event `wp_scheduled_delete` with WP Crontrol and only the promoted wasn't removed. =)

renatho committed:
Escape title HTML attribute

renatho commented:
@aaronfc, I'm merging it in order to update our CfT with this. But if you could complete the review when you can I would appreciate it, so I'll open other PRs, if needed. 😉

aaronfc commented:
> Good idea! I just did it a little different, and added a tooltip explaining the reason. I also used the data-tooltip you mentioned in the other PR. It happens in the job listing and also in the trash. WDYT?

I think it looks super cool! 🔥 

> Just to make sure, I updated the _wp_trash_meta_time in the database of one promoted job and one not promoted. So I ran the event wp_scheduled_delete with WP Crontrol and only the promoted wasn't removed. =)

Perfect! Thanks for testing it :)

In general I think this PR looks so nice! <3
